edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/ColumnFamilyDbStore.cs

```diff
@@ -75,7 +75,11 @@ public async Task Remove(byte[] key, CancellationToken cancellationToken)
 
             Action operation = () => this.db.Remove(key, this.Handle);
             await operation.ExecuteUntilCancelled(cancellationToken);
-            Interlocked.Decrement(ref this.count);
+            // We don't want to wrap around the counter.
+            if (Interlocked.Read(ref this.count) > 0)
+            {
+                Interlocked.Decrement(ref this.count);
```

Comment:
This might need to be a cmpandswap. Reading up on interlocked, but this is basically the issue

Reply:
Yeah, definitely needed to be swapped